src/v/cluster/archival/ntp_archiver_service.cc

```diff
@@ -1189,6 +1194,10 @@ bool ntp_archiver::may_begin_uploads() const {
 }
 
 ss::future<> ntp_archiver::stop() {
+    if (_stm_sub_id != archival::stm_subscriptions::id_t{}) {
```

Comment:
`_stm_sub_id != archival::stm_subscriptions::id_t{}`

:thinking: 

Is the goal here to allow all subscribers except the first to unsubscribe? If we're calling `stop()` anyways, why this condition?

Reply:
it_t{} is min int value. This checks if subscription exists.